src/python/pants/backend/jsx/target_types.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.backend.javascript.target_types import (
+    JS_FILE_EXTENSIONS,
+    JSRuntimeDependenciesField,
+    JSRuntimeSourceField,
+)
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    MultipleSourcesField,
+    OverridesField,
+    Target,
+    TargetFilesGenerator,
+    generate_file_based_overrides_field_help_message,
+    generate_multiple_sources_field_help_message,
+)
+
+JSX_FILE_EXTENSIONS = tuple(f"{ext}x" for ext in JS_FILE_EXTENSIONS)
```

Comment:
Bundlers at the very least accept and emit according to the .mjs and .cjs rule, so I think this might be a thing worth to be lenient on instead of failing?

Reply:
Happy with leniency!